### PR TITLE
device compatible, removed Variable

### PR DESCRIPTION
**Pull request by simplerick**
Changed tensor type transformations to minimal dtype transformations.
Model now compatible with any device. "to_one_hot" returns tensor of the
same device as input. Removed deprecated Variables.

### CODE REVIEW

**Comment by Pastafarianist** [comment]
Thank you for your contribution! Have you tested whether it works?

**Comment by simplerick** [comment]
Yes. I tried the model both on cpu and gpu in practice_torch.py. It works fine for me

File: week07_seq2seq/basic_model_torch.py [hunk 1]
```diff
@@ -87,9 +87,10 @@ def translate(self, inp, greedy=False, max_len = None, eps = 1e-30, **flags):
         :return: output tokens int32[batch,time] and
                  log-probabilities of all tokens at each tick, [batch,time,n_tokens]
         """
+        device = next(self.parameters()).device
```
[the remark this answers is not in the document]
Comment (simplerick) [reply week07_seq2seq/basic_model_torch.py]: image
No, as far as I know torch doesn't support this attribute by default. But if we want, we can create `self.device` in the `__init__` method.

[the remark this answers is not in the document]
Comment (simplerick) [reply week07_seq2seq/basic_model_torch.py]: But then we probably have to make another argument in `__init__` for `device`. Althought this approach has drawbacks. If I create model with CPU device and then move it to GPU by `to` method for example, `self.device` won't change and will cause error. Alternative way  to keep `self.device` correct is to define `@property` that find out actual device at each call.

[the remark this answers is not in the document]
Comment (Pastafarianist) [reply week07_seq2seq/basic_model_torch.py]: Yes, this is getting unnecessarily complex. LGTM. @justheuristic care to take a look?

**Comment by justheuristic** [comment]
LGTM. Thank you!

**Pull request merged**